common/src/main/java/com/wynntils/models/damage/DamageModel.java

```diff
@@ -76,22 +96,31 @@ public double getAverageAreaDamagePerSecond(int seconds) {
                 / (double) seconds;
     }
 
-    private final class DamageBar extends TrackedBar {
+    public final class DamageBar extends TrackedBar {
         private DamageBar() {
             super(DAMAGE_BAR_PATTERN);
         }
 
+        @Override
+        public void setEvent(LerpingBossEvent event) {
+            super.setEvent(event);
+            // Wynncraft updates the focused mob health bar by repeatedly destroying and recreating the boss bar. We
+            // grab the progress here, when the bar's recreation event is received, to avoid the brief delay before
+            // the first progress update event is received, during which the state would otherwise be inconsistent
+            onUpdateProgress(event.getProgress());
```

Comment:
This sounds like something that should be handled on a `BarHandler` (if that's how it's called) level. I'll leave this note here, and take a look, or alternatively, you can too, if you want to get this PR in faster.

Reply:
Okay, sounds sane enough.

Reply:
The comment in BloodPoolBar is just odd? I think bolyai has a point in that this would be better handled in the BossBarHandler. Can you try doing it there anyway and see if it breaks BloodPool bars? Otherwise I think this is basically a "bug fix" of BossBarHandler.

Reply:
You can always ask for help testing on Discord if you need.